aws/resource_aws_neptune_cluster_parameter_group.go

```diff
@@ -0,0 +1,263 @@
+		return fmt.Errorf("Unable to find Cluster Parameter Group: %#v", describeResp.DBClusterParameterGroups)
+	}
+
+	d.Set("name", describeResp.DBClusterParameterGroups[0].DBClusterParameterGroupName)
+	d.Set("family", describeResp.DBClusterParameterGroups[0].DBParameterGroupFamily)
+	d.Set("description", describeResp.DBClusterParameterGroups[0].Description)
+	arn := aws.StringValue(describeResp.DBClusterParameterGroups[0].DBClusterParameterGroupArn)
+	d.Set("arn", arn)
+
+	// Only include user customized parameters as there's hundreds of system/default ones
+	describeParametersOpts := neptune.DescribeDBClusterParametersInput{
+		DBClusterParameterGroupName: aws.String(d.Id()),
+		Source: aws.String("user"),
+	}
+
+	describeParametersResp, err := conn.DescribeDBClusterParameters(&describeParametersOpts)
+	if err != nil {
+		return err
+	}
+
+	if err := d.Set("parameter", flattenNeptuneParameters(describeParametersResp.Parameters)); err != nil {
+		return fmt.Errorf("error setting neptune parameter: %s", err)
+	}
+
+	resp, err := conn.ListTagsForResource(&neptune.ListTagsForResourceInput{
+		ResourceName: aws.String(arn),
+	})
+	if err != nil {
+		log.Printf("[DEBUG] Error retrieving tags for ARN: %s", arn)
+	}
+
+	if err := d.Set("tags", tagsToMapNeptune(resp.TagList)); err != nil {
+		return fmt.Errorf("error setting neptune tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsNeptuneClusterParameterGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).neptuneconn
+
+	d.Partial(true)
+
+	if d.HasChange("parameter") {
+		o, n := d.GetChange("parameter")
+		if o == nil {
+			o = new(schema.Set)
+		}
+		if n == nil {
+			n = new(schema.Set)
+		}
+
+		os := o.(*schema.Set)
+		ns := n.(*schema.Set)
+
+		parameters, err := expandNeptuneParameters(ns.Difference(os).List())
+		if err != nil {
+			return err
+		}
+
+		if len(parameters) > 0 {
+			// We can only modify 20 parameters at a time, so walk them until
+			// we've got them all.
+			for parameters != nil {
+				paramsToModify := make([]*neptune.Parameter, 0)
+				if len(parameters) <= neptuneClusterParameterGroupMaxParamsBulkEdit {
+					paramsToModify, parameters = parameters[:], nil
+				} else {
+					paramsToModify, parameters = parameters[:neptuneClusterParameterGroupMaxParamsBulkEdit], parameters[neptuneClusterParameterGroupMaxParamsBulkEdit:]
+				}
+				parameterGroupName := d.Get("name").(string)
+				modifyOpts := neptune.ModifyDBClusterParameterGroupInput{
+					DBClusterParameterGroupName: aws.String(parameterGroupName),
+					Parameters:                  paramsToModify,
+				}
+
+				log.Printf("[DEBUG] Modify Neptune Cluster Parameter Group: %s", modifyOpts)
+				_, err = conn.ModifyDBClusterParameterGroup(&modifyOpts)
+				if err != nil {
+					return fmt.Errorf("Error modifying Neptune Cluster Parameter Group: %s", err)
+				}
+			}
+			d.SetPartial("parameter")
+		}
+	}
+
+	arn := arn.ARN{
```

Comment:
We should use the value of the ARN from the Terraform state (coming from the API) rather than manually generating it:

```go
arn := d.Get("arn").(string)
```

Since we're calling update after create, we'll need to `d.Set("arn", ...)` in the create function as well as read.

Reply:
Yeah, thats what I am looking for.

Reply:
thanks, done.